bin/convertToIcns

```diff
@@ -7,18 +7,82 @@
 # ./convertToIcns ~/sample.png ~/Desktop/converted.icns
 
 # exit the shell script on error immediately
+
 set -e
 
-# import script as variable
-CONVERT_TO_PNG="${BASH_SOURCE%/*}/convertToPng"
+make_icns_sips() {
+    local file output_file iconset output_icon
+    file="${1}"
+    output_file="${2}"
+    iconset="$(mktemp -d)"
+    output_icon="$(mktemp).icns"
+
+    for size in {16,32,64,128,256,512}; do
+        sips --setProperty format png --resampleHeightWidth "${size}" "${size}" "${file}" --out "${iconset}/icon_${size}x${size}.png" &> /dev/null
+        sips --setProperty format png --resampleHeightWidth "$((size * 2))" "$((size * 2))" "${file}" --out "${iconset}/icon_${size}x${size}@2x.png" &> /dev/null
+    done
+
+    mv "${iconset}" "${iconset}.iconset"
+    iconutil --convert icns "${iconset}.iconset" --output "${output_icon}"
+
+    if [[ -z "${output_file}" ]]; then
+        echo "${output_icon}" # so its path is returned when the function ends
+    else
+        mv "${output_icon}" "${output_file}"
+    fi
+
+    # clean up
+    rm -rf "${iconset}"
+}
+
+make_icns_imagemagick() {
+    local SOURCE DEST CONVERT_TO_PNG TEMP_DIR ICONSET
+    SOURCE="${1}"
+    DEST="${2}"
+    # import script as variable
+    CONVERT_TO_PNG="${BASH_SOURCE%/*}/convertToPng"
+    TEMP_DIR="$(mktemp -d)"
+    ICONSET="$(mktemp -d)"
+
+    PNG_PATH="${TEMP_DIR}/icon.png"
+    ${CONVERT_TO_PNG} "${SOURCE}" "${PNG_PATH}"
+
+    # Resample image into iconset
+    convert "${PNG_PATH}" -define png:big-depth=16 -define png:color-type=6 -sample 16x16 "${ICONSET}/icon_16x16.png"
```

Comment:
You should only update this command. It creates jagged icons. Check out: https://github.com/webcatalog/webcatalog/issues/194

Use `-resize` instead.